WoWsShipBuilder.Web/Pages/ShipComparison.razor

```diff
@@ -0,0 +1,694 @@
+        await dataGrid.ReloadServerData();
+    }
+
+    private async Task ToggleHideShipsWithoutSelectedSection()
+    {
+        await ViewModel!.ToggleHideShipsWithoutSelectedSection();
+        UpdateUrl();
+        await dataGrid.ReloadServerData();
+    }
+
+    private async Task AddPinnedShip(ShipComparisonDataWrapper wrapper)
+    {
+        await ViewModel!.AddPinnedShip(wrapper);
+        await dataGrid.ReloadServerData();
+    }
+
+    private void AddSelectedShip(ShipComparisonDataWrapper wrapper)
+    {
+        ViewModel!.AddSelectedShip(wrapper);
+        StateHasChanged();
+    }
+
+    private async Task PinAllDisplayedShips()
+    {
+        await ViewModel!.PinAllDisplayedShips();
+        UpdateUrl();
+        await dataGrid.ReloadServerData();
+    }
+
+    private async Task ToggleShowPinnedShipOnly()
+    {
+        ViewModel!.ToggleShowPinnedShipOnly();
+        UpdateUrl();
+        await dataGrid.ReloadServerData();
+    }
+
+    private async Task ToggleUpgradedModules()
+    {
+        await ViewModel!.ToggleUpgradedModules();
+        UpdateUrl();
+        await dataGrid.ReloadServerData();
+    }
+
+    private async Task AddSearchedShip(object? obj)
+    {
+        await ViewModel!.AddShip(obj);
+        await dataGrid.ReloadServerData();
+    }
+
+    private async Task ToggleTierSelection(int tier)
+    {
+        await ViewModel!.ToggleTierSelection(tier);
+        UpdateUrl();
+        await dataGrid.ReloadServerData();
+    }
+
+    private async Task ToggleClassSelection(ShipClass shipClass)
+    {
+        await ViewModel!.ToggleClassSelection(shipClass);
+        UpdateUrl();
+        await dataGrid.ReloadServerData();
+    }
+
+    private async Task ToggleNationSelection(Nation nation)
+    {
+        await ViewModel!.ToggleNationSelection(nation);
+        UpdateUrl();
+        await dataGrid.ReloadServerData();
+    }
+
+    private async Task ToggleCategorySelection(ShipCategory category)
+    {
+        await ViewModel!.ToggleCategorySelection(category);
+        UpdateUrl();
+        await dataGrid.ReloadServerData();
+    }
+
+    private async Task ToggleAllCategories(bool toggle)
+    {
+        await ViewModel!.ToggleAllCategories(toggle);
+        UpdateUrl();
+        await dataGrid.ReloadServerData();
+    }
+
+    private async Task ToggleAllClasses(bool toggle)
+    {
+        await ViewModel!.ToggleAllClasses(toggle);
+        UpdateUrl();
+        await dataGrid.ReloadServerData();
+    }
+
+    private async Task ToggleAllNations(bool toggle)
+    {
+        await ViewModel!.ToggleAllNations(toggle);
+        UpdateUrl();
+        await dataGrid.ReloadServerData();
+    }
+
+    private async Task ToggleAllTiers(bool toggle)
+    {
+        await ViewModel!.ToggleAllTiers(toggle);
+        UpdateUrl();
+        await dataGrid.ReloadServerData();
+    }
+
+    private string GetSelectAllButtonText()
+    {
+        return ViewModel!.SelectAllShips ? "Unselect All" : "Select All";
```

Comment:
Localization

Reply:
Currently the entire feature is missing localization. it will be added during the beta testing